src/forest.py

```
@@ -389,6 +419,14 @@ def adding_isolines_by_height(
     Добавляет в новый слой только те изолинии, высота которых попадает в заданный диапазон.
     """
     for feature in contours_layer.getFeatures():
+        if progress.was_canceled():
+            return
+
+        processed += 1
```

Ruff (0.11.9)

Undefined name `processed`

(F821)